core/lib/dal/src/blocks_web3_dal.rs

```diff
@@ -420,32 +420,45 @@ impl BlocksWeb3Dal<'_, '_> {
         Ok(hash)
     }
 
-    pub async fn get_l2_to_l1_logs(
+    pub async fn get_l2_to_l1_logs_by_number<L>(
```

Comment:
IMO, it would make sense to make the templated method crate-private and expose its non-templated public variant with the same signature as before:

```rust
pub(crate) async fn get_l2_to_l1_logs_by_number<L>(
    &mut self,
    l1_batch_number: L1BatchNumber,
) -> DalResult<Vec<L>>
where
    L: From<StorageL2ToL1Log>,
{
    // ...
}

pub async fn get_l2_to_l1_logs(
    &mut self,
    l1_batch_number: L1BatchNumber,
) -> DalResult<Vec<api::L2ToL1Log>> {
    self.get_l2_to_l1_logs_by_number(l1_batch_number).await
}
```

It may make sense to place the crate-private method in `BlocksDal` since it is used outside of Web3-related methods.

Reply:
fair. Thanks for the suggestion.